packages/field-plugin/src/messaging/pluginMessage/containerToPluginMessage/MessageToPlugin.ts

```diff
@@ -3,6 +3,7 @@ import { hasKey } from '../../../utils'
 export type MessageToPlugin<Action extends string> = {
   action: Action
   uid: string
+  callbackId?: string
```

Comment:
Every message has potentially a `callbackId`. It is optional because sometimes, the event was not triggered by the plugin, so there would be no callback to refer to

Reply:
Can you give an example?

Reply:
The following `MessageToContainer` messages resolve a promise: loaded, `asset-selected`, `update`, `toggle-modal`, `request-context`. Therefore, all these messages can contain a `callbackId`.

The `heightChange` message does not resolve a promise. Right now, I don't see a reason why it would. However, it is conceivable. Not sure if we should omit `callbackId` from `MessageToContainer` because of this.

For `MessageToPlugin`, all messages could resolve a promise: `loaded`, `state-changed`, `asset-selected`, `context`, so all can contain a `callbackId`.

Actually, right now, there is no real example of a `MessageToPlugin` message that does _not_ contain a `callbackId`... but, I can invent one!

Let's say the user clicks outside the modal window: ideally, this would close the modal window. Upon such click event, the Visual Editor sends a `state-updated` message to the field plugin. The event originated from the Visual Editor, so there is no promise to resolve, therefore the `callbackId` is omitted.

Reply:
Ah, thanks for the explanation. Yeah it makes sense to have it optional. We cannot revert it if we being with it being required :)

Reply:
@johannes-lindgren @eunjae-lee Can you explain what how we will use the callbackId, or why does it need to be present?